rivertype/river_type.go

```diff
@@ -303,3 +306,19 @@ type Queue struct {
 	// deleted from the table by a maintenance process.
 	UpdatedAt time.Time
 }
+
+// UniqueOptsByStateDefault is the set of job states that are used to determine
+// uniqueness unless unique job states have been overridden with
+// UniqueOpts.ByState. So for example, with this default set a new unique job
+// may be inserted even if another job already exists, as long as that other job
+// is set `cancelled` or `discarded`.
+func UniqueOptsByStateDefault() []JobState {
```

Comment:
Naming on this one seems a little awkward, but I think it might be the best way to go: it's basically `UniqueOpts` + `ByState` + `Default` so it copies the taxonomy of the field it's being set to.